### ISSUE TITLE
Nan and other failure cases handelling

### ISSUE THREAD

**Issue by shubham-shahh** [ArduPilot/ardupilot#19382]
## Bug report

Hi, I have described the issue[ here](https://github.com/mavlink/mavros/issues/1647)

_Please describe the problem_

The copter tends to crash when a path planner passes "Nan" values. 
**Version**
_What version was the issue encountered with_
4.0.5
**Platform**
[  ] All
[  ] AntennaTracker
[X] Copter
[  ] Plane
[  ] Rover
[  ] Submarine

**Airframe type**
Quad

**Hardware type**
Pixhawk (SITL)

**Logs**
NA

**Comment by magicrub** [comment]
Can you be more specific?

**Comment by shubham-shahh** [comment]
Hi,
the external path planners sometimes give "nan" values for yaw-rate (for example fast planner) if we convert that message to mavros type and pass it to ardupilot the copter firmware crashes (tested it in SITL).
I don't have logs but if logs are required I'll reproduce the issue once again

thanks

**Comment by rmackay9** [comment]
@shubham-shahh which mavlink messages in particular are you referring to?  Here are some choices:

- SET_POSITION_TARGET_LOCAL_NED
- SET_POSITION_TARGET_GLOBAL_INT
- SET_ATTITUDE_TARGET

**Comment by shubham-shahh** [comment]
> @shubham-shahh which mavlink messages in particular are you referring to? Here are some choices:
> 
> * SET_POSITION_TARGET_LOCAL_NED
> * SET_POSITION_TARGET_GLOBAL_INT
> * SET_ATTITUDE_TARGET

Hi, @rmackay9, I am using /mavros/setpoint_raw/local of type mavros_msgs::PositionTarget which refers to the mavlink message SET_POSITION_TARGET_LOCAL_NED

**Comment by shubham-shahh** [comment]
> @shubham-shahh which mavlink messages in particular are you referring to? Here are some choices:
> 
> * SET_POSITION_TARGET_LOCAL_NED
> * SET_POSITION_TARGET_GLOBAL_INT
> * SET_ATTITUDE_TARGET

Hi, whats your opinion? shall we add such failure cases in ardupilot? if yes Please let me know, I would love to contribute more @rmackay9

**Comment by rmackay9** [comment]
@shubham-shahh, yes, I think if the field is used we should check for NaN and reject the message if it contains one (or more).

**Comment by shubham-shahh** [comment]
> @shubham-shahh, yes, I think if the field is used we should check for NaN and reject the message if it contains one (or more).

Hi, thanks for the feedback. I was curious if we should do this with all MAVLINK messages?

I'll work on the correction and put up a PR.

Thanks

**Comment by shubham-shahh** [comment]
> @shubham-shahh, yes, I think if the field is used we should check for NaN and reject the message if it contains one (or more).

I tired to re create the issue to solve it but looks like it is fixed

**Comment by shubham-shahh** [comment]
> 

I was planning to add a fix [here](https://github.com/ArduPilot/ardupilot/blob/308f4e99b4d3cda98ad89529bfc3d3faaa800173/ArduCopter/GCS_Mavlink.cpp#L1217)
